### PR TITLE
[GOBBLIN-1920] Use db laundered timestamp for reminder event

### PR DESCRIPTION
**Pull request by umustafi**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [X] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1920

### Description
- [X] Here are some details about my PR, including screenshots (if applicable):
When setting the reminder event we use the trigger time local to each host. Instead we want to use the timestamp from database to see consistency between hosts in case their clock drift is significant. 

### Tests
- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [X] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Review by phet — APPROVED** [review]
good catch!

only improvement to recommend is a comment explaining why this is important

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/FlowTriggerHandler.java [hunk 1]
```diff
@@ -277,8 +277,9 @@ public static JobDataMap updatePropsInJobDataMap(JobDataMap jobDataMap,
     // Saves the following properties in jobProps to retrieve when the trigger fires
     prevJobProps.setProperty(ConfigurationKeys.SCHEDULER_EXPECTED_REMINDER_TIME_MILLIS_KEY,
         String.valueOf(getUTCTimeFromDelayPeriod(delayPeriodMillis)));
+    // We use the db laundered timestamp for the reminder to ensure consensus between hosts  
     prevJobProps.setProperty(ConfigurationKeys.SCHEDULER_PRESERVED_CONSENSUS_EVENT_TIME_MILLIS_KEY,
-        String.valueOf(triggerEventTimeMillis));
```
Comment (prasadkarkera) [inline gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/FlowTriggerHandler.java:281]: We can probably remove this from function parameter as well?

**Review by meethngala — COMMENTED** [review]
nice catch! I was wondering if it even makes sense for us to provide `eventTimeMillis` to the method signature of `scheduleReminderForEvent`. Earlier we were using it to create `JobDetail`, but now the only place we use it is for logging and the info that it conveys is the clock drift (which we might be okay not logging)... let me know what's your take?

**Comment by umustafi** [comment]
> nice catch! I was wondering if it even makes sense for us to provide `eventTimeMillis` to the method signature of `scheduleReminderForEvent`. Earlier we were using it to create `JobDetail`, but now the only place we use it is for logging and the info that it conveys is the clock drift (which we might be okay not logging)... let me know what's your take?

I am leaving it in to catch that clock drift in time being. I suspect it's not the root of our problem but until we are reliably bug free keeping logs in.

**Review by Will-Lo — APPROVED** [review]

**Pull request merged**